phonenumber_field/modelfields.py

```diff
@@ -77,16 +77,24 @@ def get_prep_value(self, value):
         """
         Perform preliminary non-db specific value checks and conversions.
         """
-        if value:
-            if not isinstance(value, PhoneNumber):
-                value = to_python(value)
+        if not value:
+            return super().get_prep_value(value)
 
-            if not value.is_valid():
-                raise ValueError("“%s” is not a valid phone number." % value.raw_input)
+        if isinstance(value, PhoneNumber):
+            parsed_value = value
+        else:
+            # Convert the string to a PhoneNumber object.
+            parsed_value = to_python(value)
```

Comment:
Not sure if this line won't raise an error in case of an invalid number, but other than this, LGTM.

Reply:
As long as phonenumbers does as promised and raises a `NumberParseException` if something is wrong with the number we should be fine. In `to_python` all numbers that cause a `NumberParseException` are stored in their raw format.
https://github.com/stefanfoulis/django-phonenumber-field/blob/cdb3d1251a0c4de0f889a249c141e20aec167c7e/phonenumber_field/phonenumber.py#L115-L117